newton/_src/sensors/tiled_camera_sensor.py

```diff
@@ -0,0 +1,491 @@
+                model.shape_flags,
+                self.render_context.geom_enabled,
+                self.render_context.geom_mesh_indices,
+                num_enabled_geoms,
+            ],
+        )
+        self.render_context.num_geom_in_bvh = int(num_enabled_geoms.numpy()[0])
+
+        wp.launch(
+            kernel=compute_mesh_bounds,
+            dim=self.model.shape_count,
+            inputs=[self.render_context.mesh_ids, self.render_context.mesh_bounds],
+        )
+
+    def update_from_state(self, state: State):
+        """
+        Update data from Newton State.
+
+        Args:
+            state: The current simulation state containing body transforms.
+        """
+        if self.render_context.has_geometries:
+            wp.launch(
+                kernel=convert_newton_transform,
+                dim=self.model.shape_count,
+                inputs=[
+                    state.body_q,
+                    self.model.shape_body,
+                    self.model.shape_transform,
+                    self.model.shape_scale,
+                    self.render_context.geom_positions,
+                    self.render_context.geom_orientations,
+                    self.render_context.geom_sizes,
+                ],
+            )
+
+        if self.render_context.has_triangle_mesh:
+            self.render_context.triangle_points = state.particle_q
+
+        if self.render_context.has_particles:
+            self.render_context.particles_position = state.particle_q
+
+    def render(
+        self,
+        color_image: wp.array(dtype=wp.uint32, ndim=3) | None = None,
+        depth_image: wp.array(dtype=wp.float32, ndim=3) | None = None,
+        refit_bvh: bool = True,
+        clear_images: bool = True,
+    ):
+        """
+        Render color and depth images for all worlds and cameras.
+
+        Args:
+            color_image: Optional output array for color data (num_worlds, num_cameras, width*height).
+                        If None, no color rendering is performed.
+            depth_image: Optional output array for depth data (num_worlds, num_cameras, width*height).
+                        If None, no depth rendering is performed.
+            refit_bvh: Whether to rebuild the BVH or not.
+            clear_images: Whether to clear the images before rendering or not.
+        """
+        self.render_context.render(color_image, depth_image, refit_bvh=refit_bvh, clear_images=clear_images)
+
+    def update_cameras(self, positions: wp.array(dtype=wp.vec3f), orientations: wp.array(dtype=wp.mat33f)):
+        """
+        Update camera positions and orientations.
+
+        Args:
+            positions: Array of camera positions in world space, shape (num_cameras,).
+            orientations: Array of camera-to-world rotation matrices (as flattened 3x3), shape (num_cameras,).
+        """
+        self.render_context.camera_positions = positions
+        self.render_context.camera_orientations = orientations
```

Comment:
We use `wp.transform` usually in Newton for representing transforms rather than storing positions and orientations separately.